### ISSUE TITLE
No file lock for BondiSachs interpolation target

### ISSUE THREAD

**Issue by knelli2** [sxs-collaboration/spectre#4966]
In `DumpBondiSachsOnWorldtube`, I do

```cpp
            observers::ThreadedActions::ReductionActions_detail::write_data(
                subfile_name, observers::input_source_from_cache(cache), legend,
                std::make_tuple(data_to_write_buffer), filename,
                std::index_sequence<0>{});
```

which doesn't go through the `ObserverWriter` so I think this is causing some H5 issues on Ocean (and maybe Wheeler) because two processes might be trying to write to the file at the same time.

One reason this method is nice is because it (effectively) ensures that the data is written in order (monotonically increasing) which I think is necessary for CCE. However, we should go through the `ObserverWriter` so we don't get these h5 errors, but now the output data won't be in order so we need something to keep it in order.

So there's two things I think that need to happen:

- [ ] Use `ObserverWriter` in `DumpBondiSachsOnWorldtube`
- [ ] Write python tool to order the output of a dat file in an h5 file

**Comment by nilsvu** [comment]
Won't the output be in order if you send it to the writer in order?

**Comment by knelli2** [comment]
No? I thought charm doesn't guarantee the order in which messages arrive, even if they are sent sequentially.

**Comment by nilsvu** [comment]
Hm the reduction data I've seen so far has been ordered, but maybe it's not guaranteed. I'd try it and deal with sorting later if you need to.

Edit: with that I mean, send your data to the ObserverWriter in a simple action and write from there.

**Comment by knelli2** [comment]
I don't think it's guaranteed. If you write infrequent enough, it'll be ordered though. For the control system, I occasionally see data written out of order.

That being said, I now don't think using `write_data` here is a problem. `DumpBondiSachsOnWorldtube` doesn't write to the reductions, surface, or volume file at all. It writes to it's own files that are in the format CCE needs. So if this is the only function that ever writes to those files, then I don't think there should be any errors.

**Comment by knelli2** [comment]
Fixed in #5300
